api/v1beta1/kustomization_types.go

```diff
@@ -184,6 +185,15 @@ func KustomizationProgressing(k Kustomization) Kustomization {
 	return k
 }
 
+// SetKustomizationHealthiness sets the HealthyCondition status for a Kustomization.
+func SetKustomizationHealthiness(k *Kustomization, status metav1.ConditionStatus, reason, message string) {
+	if len(k.Spec.HealthChecks) > 0 {
+		meta.SetResourceCondition(k, HealthyCondition, status, reason, trimString(message, MaxConditionMessageLength))
+	} else {
+		apimeta.RemoveStatusCondition(k.GetStatusConditions(), HealthyCondition)
+	}
```

Comment:
Maybe cleaner to always remove and then conditionally add back?

Reply:
I don't think this is the intended usage of the apimeta. If a condition exists, we should update it instead of removing and creating it from scratch.